src/main/groovy/edu/oregonstate/mist/locations/frontend/resources/LocationResource.groovy

```diff
@@ -0,0 +1,261 @@
+package edu.oregonstate.mist.locations.frontend.resources
+
+import com.codahale.metrics.annotation.Timed
+import com.fasterxml.jackson.databind.ObjectMapper
+import edu.oregonstate.mist.api.AuthenticatedUser
+import io.dropwizard.auth.Auth
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.WebApplicationException
+import javax.ws.rs.core.Context
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.MultivaluedMap
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.UriInfo
+import java.util.regex.Pattern
+
+@Path("/locations")
+@Produces(MediaType.APPLICATION_JSON)
+class LocationResource {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocationResource.class)
+    public static final ArrayList<String> ALLOWED_CAMPUSES = ["corvallis"]
+    public static final ArrayList<String> ALLOWED_TYPES = ["building", "dining"]
+
+    /**
+     * Default page number used in pagination
+     * @todo: should this be coming from skeleton or configuration file?
+     */
+    public static final Integer DEFAULT_PAGE_NUMBER = 1
+
+    /**
+     * Default page size used in pagination
+     * @todo: should this be coming from skeleton or configuration file?
+     */
+    public static final Integer DEFAULT_PAGE_SIZE = 10
```

Comment:
if you want to be able to set the paging parameters at execution time, it would make sense to read from the configuration file. if you do not expect the paging behavior to change between executions, either define the parameters here or in a different file in `edu.oregonstate.mist.locations`.

i would not recommend modifying the skeleton until this api has implemented one of those options. note that it may not be trivial to implement paging functionality in the skeleton due to the differences between various apis.